debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java

```diff
@@ -346,8 +345,6 @@ public MySqlConnectorConfig(Configuration config) {
         final String gtidSetExcludes = config.getString(GTID_SOURCE_EXCLUDES);
         this.gtidSourceFilter = gtidSetIncludes != null ? Predicates.includesUuids(gtidSetIncludes)
                 : (gtidSetExcludes != null ? Predicates.excludesUuids(gtidSetExcludes) : null);
-
-        getServiceRegistry().registerServiceProvider(new MySqlCharsetRegistryServiceProvider());
```

Comment:
I suspect if we reverse course on the `MariaDbCharsetRegistry` behavior, it would make sense to do that here as well; although unlikely MariaDB, the MySQL registry doesn't parse anything and acts as a delegate to the driver.

Reply:
Apologies, I didn’t fully understand your comment, could you please elaborate?

In this case, I removed the load call because I assumed it was intended to eagerly load the `MySqlCharsetRegistryServiceProvider`, which I’ve now moved inside the `ConnectorTask` (please see the comment in [MySqlConnectorTask.java](https://github.com/debezium/debezium/pull/6458/files#diff-2fb991f953c491e9f94e023c0f9c1cf0d910594df1d09fcb457a76e97b1409fd)).

Reply:
Sorry that my comment wasn't clear. I was conveying that if we chose to revert back so that there is just one `BinlogCharsetRegistry` instance created , it would make sense to align the MariaDB/MySQL behavior despite the fact that using different instances does not have IO concerns like MariaDB's implementation does.